### DISCUSSION TITLE
Will Bevy ever “Annex” a physics engine?

### DISCUSSION THREAD

**Discussion by Adamkob12** [bevyengine/bevy#10709] [Q&A]
Bevy xpbd fits Bevy like a glove. It was built as a 3rd party crate so it's as modular as it gets, the community loves it, and no game engine is complete without a built in physics engine.

If bevy takes xpbd under its wing, we can accelerate its development.

**Comment by alice-i-cecile** [comment]
My opinion (maintainer, but only speaking for myself):

1. Bevy should eventually have its own physics engine.
2. We shouldn't build our own from scratch.
3. We shouldn't include nalgebra in our tree: the duplicate math crates are complex and easy to screw up. So bevy_xpbd will need its own collision backend.
4. bevy_xpbd is well-made, and fits great with how Bevy wants to work.
5. The xpbd part of bevy_xpbd might be a bit dicey WRT patents, so I would either want an explicit blessing or an alternative solver (with the ability to swap out parts of the physics engine freely).

**Comment by amytimed** [comment]
as a bevy_rapier enjoyer i am curious why bevy_xpbd specifically would be chosen

bevy_rapier has almost the same API and thus "fits bevy like a glove" in the same way, but has more github stars and activity

**Comment by mogambro** [comment]
> as a bevy_rapier enjoyer i am curious why bevy_xpbd specifically would be chosen
> 
> bevy_rapier has almost the same API and thus "fits bevy like a glove" in the same way, but has more github stars and activity

That's @alice-i-cecile opinion, but she stated: 

> "We shouldn't include nalgebra in our tree: the duplicate math crates are complex and easy to screw up. So bevy_xpbd will need its own collision backend."

`rapier` builds upon `nalgebra`
